### DISCUSSION TITLE
Proposed poll: Why do you use Radicale?

### DISCUSSION THREAD

**Discussion by tmst** [Kozea/Radicale#1385] [Show and tell]
I'd like to find out what motivates people to run their own CalDAV/Card server and how they run it, etc. Wouldn't that be interesting?

If you can think of some other questions, or know of a good polling website please let me know.

Here's my initial thoughts:

What best describes your reason for running your own Cal/CardDAV server?
- Privacy
- Control
- Price
- Rebellion

How do you commonly run Radicale?
- Local network
- AWS

What best describes your choice of Radicale?
- Great development team
- Understandable implementation
- Written in Python

**Comment by lsergei** [comment]
1. Privacy
2. Own servers on DigitalOcean
3. Great team and understandable implementaton

**Comment by tmst** [comment]
On Mon, Feb 26, 2018, at 7:45 AM, lsergei wrote:
>  1. Privacy
>  2. Own servers on DigitalOcean
>  3. Great team and understandable implementation
Yeah. Like this.

**Comment by tmst** [comment]
On Mon, Feb 26, 2018, at 5:15 PM, Tom Russell wrote:
> On Mon, Feb 26, 2018, at 7:45 AM, lsergei wrote:
>>  1. Privacy
>>  2. Own servers on DigitalOcean
>>  3. Great team and understandable implementation> 
> Yeah. Like this.

Though I have to admit that even having mastered the fundamentals of
Python programming and having worked on a large client-server
application in C++ my eyes still go crossed when I try to understand how
Radicale works. One of these days I'm going to give it a real college
try. Is there any documentation, outside of source comments, of the
control flow from request through response?

**Comment by MacLemon** [comment]
We were using Apple `calendarserver` on macOS Server which has been working fine. Recently Apple (finally, not unexpectedly) deprecated macOS server.

 We've tried a ton of different FOSS CalDAV solutions over a long period of time and pretty much nothing worked properly (including Radicale 1.x back then). Since we can't just go eff it and use (Google Calendar|iCloud|Exchange|MS Outlook) we needed a self-hosting solution that at least would let us sync multiple collections per user across multiple devices and offer some, even if absolutely rudimentary, group calendar functionality.

Radicale 2.1.8 was the first release we tried in the 2.x tree and we got it to work. We're lacking much functionality over Apple's calendar server like invitations, scheduling, delegations, but it get's the bare minimum done. (Unlike many other CalDAV implementations.)
So for small installations that just need some calendars and maybe sync an Addressbook across their laptop and phone, Radicale is a really fine solution.

Things that are better than Apple CalendarServer:  
We did gain somewhat better granularity on r/w privileges (though they have to be administered by editing a file which is highly error prone and mistakes that leak data are very easy to make.)
We also did gain the ability to have some collections also be available as a public subscription.

We're looking into a more full fledged solution in the meantime and have high hopes for [Cyrus IMAPd](https://cyrusimap.org/imap/download/installation/http/caldav.html?highlight=caldav#caldav) since that one actually covers pretty much all the CalDAV things we're missing from Radicale. Cyrus scales to thousands of users if need be.

**Comment by hakova** [comment]
1. Privacy
2. Local network
3. Understandable implementation seems to be the best option here. In my own words, the answer is that  it was simple enough to setup and use, yet robust.

**Comment by SimplyDanny** [comment]
I use it because I have full control over the events within calendars on the server side. So it is possible to add events automatically after first collecting them from other sources (e.g. web pages) and keep them up-to-date.

**Comment by tmst** [comment]
On Wed, Apr 11, 2018, at 11:26 AM, Danny Mösch wrote:
> I use it because I have full control over the events within calendars
> on the server side. So it is possible to add events automatically
> after first collecting them from other sources (e.g. web pages) and
> keep them up-to-date.
Interesting. I'm not sure I understand how this differs from manually
adding an event using a web or other client. Do you add events to the
server's file system directly?

**Comment by tmst** [comment]
I just spotted #775 with your answer to my question. Very nice.

**Comment by firestorm99** [comment]
What best describes your reason for running your own Cal/CardDAV server?

   --> Privacy
    Control
    Price
    Rebellion

How do you commonly run Radicale?

    Local network
    AWS
   --> KVM VPS

What best describes your choice of Radicale?

    Great development team
    Understandable implementation
    Written in Python
  --> Easy to configure, built-in https-support, no extra webserver needed.
